packages/@aws-cdk/aws-pipes-alpha/lib/inputTransformation.ts

```diff
@@ -0,0 +1,55 @@
+
+enum reservedVariables {
+  PIPES_ARN = '<aws.pipes.pipe-arn>',
+  PIPES_NAME = '<aws.pipes.pipe-name>',
+  PIPES_TARGET_ARN = '<aws.pipes.target-arn>',
+  PIPE_EVENT_INGESTION_TIME = '<aws.pipes.event.ingestion-time>',
+  PIPE_EVENT = '<aws.pipes.event>',
+  PIPE_EVENT_JSON = '<aws.pipes.event.json>',
+}
+
+type StaticString = string;
+// type JsonPath = `<$.${string}>`;
+type KeyValue = Record<string, string | reservedVariables>;
+type StaticJsonFlat = Record<string, StaticString | KeyValue>;
+type InputTransformJson = Record<string, StaticString | KeyValue | StaticJsonFlat>;
+
+type InputTransformationValue = StaticString | InputTransformJson;
+
+/**
+ * Transform or replace the input event payload
+ */
+export interface IInputTransformation {
+  /**
+   * Valid JSON text passed to the target.
+   */
+  inputTemplate: string;
+}
+
+/**
+ * Transform or replace the input event payload
+ */
+export class InputTransformation implements IInputTransformation {
+  /**
+   * Builds an input transformation from a JSON object.
+   * @param inputTemplate
+   * @returns
+   */
+  static fromJson(inputTemplate: Record<string, any>): InputTransformation {
+    return new InputTransformation(inputTemplate);
+  }
+
+  inputTemplate: string;
+
+  constructor(inputTemplate: InputTransformationValue) {
```

Comment:
Should this be a private constructor? If not, we have a problem with the types further up.

Reply:
I will have another look at it. 
I think the inputTransformation should be build similar to the `RuleTargetInput` class from the EventBridge rules. 
https://github.com/aws/aws-cdk/blob/v2.117.0/packages/aws-cdk-lib/aws-events/lib/input.ts#L10

But I have to get my head around the implementation there first.